src/modules/auth/component/SignUp/SignUp.tsx

```diff
@@ -1,58 +1,62 @@
+/* eslint-disable no-console */
 import {
   Grid,
+  Typography,
 } from "@mui/material";
-import {
-  ChangeEvent, FormEvent, useEffect, useState,
-} from "react";
+import { useEffect } from "react";
 import { v4 as uuidv4 } from "uuid";
+import { Controller, SubmitHandler, useForm } from "react-hook-form";
 import { PrimaryButton } from "../../../../components/UI/PrimaryButton";
 import { TextField } from "../../../../components/UI/TextField";
 import { Heading } from "../../../../components/UI/Heading";
-import { profileData, token } from "../../../../data-mock/data";
+import { token } from "../../../../data-mock/data";
 import { Profile } from "../../../../types/types";
 import { useAppDispatch } from "../../../../store/hooks";
 import { login } from "../../../../store/features/tokenSlice";
-import { UserRepository, userRepository } from "../../../../data-mock/userMock";
+import { User, UserRepository, userRepository } from "../../../../data-mock/userMock";
 
 export function SignUp() {
-  const [formData, setFormData] = useState<Omit<Profile, "id" | "isLoggedIn">>(
-    profileData,
-  );
+  const {
+    control, setError, handleSubmit, formState: { errors }, reset, getValues,
```

Comment:
Do not use formState: { errors } like in the documentation. It is not a very good practice.
use this approach:
```
const {
    control, 
    setError, 
    handleSubmit, 
    formState, 
    reset, 
    getValues, 
    ...form
} = useForm({
    defaultValues: {
      name: "",
      email: "",
      password: "",
      confirmPassword: "",
    },
  });

const { errors } = formState;
```

Reply:
done